Assets/VRTK/Source/Scripts/Utilities/VRTK_SharedMethods.cs

```diff
@@ -472,6 +469,109 @@ public static Vector3 AxisDirection(int axisIndex, Transform givenTransform = nu
             return worldDirections[(int)Mathf.Clamp(axisIndex, 0f, worldDirections.Length)];
         }
 
+        /// <summary>
+        /// The AddListValue method attempts to add a value in the given list if the value does not already exist. `forceAdd = true` will always add a new value to the list even if it already exists.
+        /// </summary>
+        /// <typeparam name="TValue">The datatype for the list value.</typeparam>
+        /// <param name="list">The list to retrieve the value from.</param>
+        /// <param name="value">The value to attempt to add to the list.</param>
+        /// <param name="forceAdd">If this is `true` then the value provided will always be appended to the list. If this is `false` the value provided will only be added to the list if it doesn't already exist.</param>
+        /// <returns>Returns `true` if the given value was successfully added to the list. Returns `false` if the given value already existed in the list and `forceAdd = false`.</returns>
+        public static bool AddListValue<TValue>(ref List<TValue> list, TValue value, bool forceAdd = false)
+        {
+            if (forceAdd || !list.Contains(value))
```

Comment:
Use `HashSet<TValue>` if you need to ensure there are no duplicated items, don't use a `List<TValue>`.

`HashSet<TValue>` has way better performance.

If Unity doesn't offer a default inspector for `HashSet<TValue>`s we can provide one.

Reply:
iirc hash sets are only more performant at higher record counts, and lower records a list should outperform a hashset

Reply:
That may be true but the difference is either negligible or the added readability and transparency on the handling of duplicates is worth it in my opinion.

Reply:
But then you have to build a custom property drawer for Unity as it doesn't support HashSets.

It's a wibbly wobbly world of it's own! :)